source/AI.cpp

```diff
@@ -119,10 +119,74 @@ namespace {
 	void Deploy(const Ship &ship, bool includingDamaged)
 	{
 		for(const Ship::Bay &bay : ship.Bays())
-			if(bay.ship && (includingDamaged || bay.ship->Health() > .75))
+			if(bay.ship && (includingDamaged || bay.ship->Health() > .75) &&
+					(!bay.ship->IsYours() || bay.ship->IsDeploying()))
 				bay.ship->SetCommands(Command::DEPLOY);
 	}
 	
+	// Issue deploy orders for the selected ships (or the full fleet if no ships are selected).
+	void IssueDeploy(const PlayerInfo &player)
+	{
+		// Figure out what carried ships we are giving orders to
+		vector<Ship *> ships;
+		bool fullFleet = player.SelectedShips().empty();
+		if(fullFleet)
+		{
+			for(const shared_ptr<Ship> &it : player.Ships())
+				if(it.get() != player.Flagship() && !it->IsParked() && it->CanBeCarried())
+					ships.push_back(it.get());
+		}
+		else
+			for(const weak_ptr<Ship> &it : player.SelectedShips())
+			{
+				shared_ptr<Ship> ship = it.lock();
+				if(ship)
+					ships.push_back(ship.get());
+			}
+		
+		// This should never happen, but just in case:
+		if(ships.empty())
+		{
+			if(fullFleet)
+				Messages::Add("No carried ships in the fleet to deploy.");
+			else
+				Messages::Add("No carried ships selected that can be deployed.");
```

Comment:
This is currently unreachable, due to the above missing condition:
![image](https://user-images.githubusercontent.com/20871346/86387285-96b7e700-bc58-11ea-8cbb-f40ba3f302b6.png)

Reply:
Agreed, removed the unreachable message.